Hazel/Crypto/AesGcm.cs

```diff
@@ -322,48 +322,59 @@ static void MultiplyGF128Elements(ByteSpan X, ByteSpan Y, ByteSpan scratchZ, Byt
             //   irreducible polynomial is represented by the bit string:
             //   `11100001` followed by 120 `0`s. We can add this value to `V`
             //   by: `V[0] = V[0] ^ 0xE1`.
-            SetSpanToZeros(scratchZ);
+            scratchZ.SecureClear();
             X.CopyTo(scratchV);
 
-            for (int ii = 0; ii != 128; ++ii)
+            unchecked
             {
-                int bitIndex = 7 - (ii % 8);
-                if ((Y[ii / 8] & (1 << bitIndex)) != 0)
+                fixed (byte* scratchZPtr = &scratchZ.GetUnderlyingArray()[scratchZ.Offset])
+                fixed (byte* scratchVPtr = &scratchV.GetUnderlyingArray()[scratchV.Offset])
+                fixed (byte* YPtr = &Y.GetUnderlyingArray()[Y.Offset])
                 {
-                    for (int jj = 0; jj != 16; ++jj)
+                    for (int ii = 0; ii != 128; ++ii)
                     {
-                        scratchZ[jj] ^= scratchV[jj];
+                        int bitIndex = 7 - (ii % 8);
+                        if ((YPtr[ii / 8] & (1 << bitIndex)) != 0)
+                        {
+                            scratchZPtr[0] ^= scratchVPtr[0];
```

Comment:
These kinds of bitwise ops and the SecureClear method make me wonder if bringing in System.Numerics and using Vector<T> could be helpful for perf: [Vector<T> Struct](https://learn.microsoft.com/en-us/dotnet/api/system.numerics.vector-1?view=netstandard-2.1)

Also found some examples: [https://github.com/CBGonzalez/SIMDIntro](https://github.com/CBGonzalez/SIMDIntro)

I haven't used it personally, but does seem like this is the sort of use case for it, so maybe that would be next steps if we find we still need perf improvements?

Reply:
Yeah! That's neat. We can def do some perf testing and see if it helps.